### PR TITLE
Release version 2.22.0: "E-Bike Excitement"

### PR DESCRIPTION
**Pull request by rcurtin**
This automatically-generated pull request adds the commits necessary to make the 2.22.0 release.

Once the PR is merged, mlpack-bot will tag the release as HEAD~1 (so that it doesn't include the new HISTORY block) and publish it.

Or, well, hopefully that will happen someday.

When you merge this PR, be sure to merge it using a *rebase*.

### Changelog

 * Update to C++14 standard ([#400](https://github.com/mlpack/ensmallen/pull/400)).

 * Bump minimum Armadillo version to 10.8 ([#404](https://github.com/mlpack/ensmallen/pull/404)).

 * For Armadillo 14.2.0 switch to `.index_min()` and `.index_max()` ([#409](https://github.com/mlpack/ensmallen/pull/409).

### CODE REVIEW

**Review by coatless — APPROVED** [review]
LGTM!

**Review by conradsnicta — APPROVED** [review]

**Comment by rcurtin** [comment]
I will actually wait until #403 is merged because it's approved now---so I'll have to make a new release PR that includes that.

**Comment by conradsnicta** [comment]
@rcurtin #403 has been merged

**Pull request closed without merging**
